src/LocalDatastoreController.react-native.js

```diff
@@ -27,29 +28,63 @@ const LocalDatastoreController = {
       await RNStorage.setItemAsync(name, values);
     } catch (e) {
       // Quota exceeded, possibly due to Safari Private Browsing mode
-      console.log(e.message); // eslint-disable-line no-console
+      console.error(e.message);
     }
-    return Promise.resolve();
   },
 
-  async unPinWithName(name: string): Promise {
-    await RNStorage.removeItemAsync(name);
-    return Promise.resolve();
+  unPinWithName(name: string): Promise {
+    return RNStorage.removeItemAsync(name);
   },
 
   async getAllContents(): Promise {
-    const LDS = {};
     const keys = await RNStorage.getAllKeys();
+    const batch = [];
     for (let i = 0; i < keys.length; i += 1) {
       const key = keys[i];
-      const value = await RNStorage.getItemAsync(key);
-      LDS[key] = JSON.parse(value);
+      if (isLocalDatastoreKey(key)) {
+        batch.push(key);
+      }
+    }
+    const LDS = {};
+    let results = [];
+    try {
+      results = await RNStorage.multiGet(batch);
+    } catch (error) {
+      console.error('Error getAllContents: ', error);
+      return {};
     }
-    return Promise.resolve(LDS);
+    results.map((pair) => {
+      const [key, value] = pair;
+      try {
+        LDS[key] = JSON.parse(value);
+      } catch (error) {
+        LDS[key] = null;
+      }
+    });
+    return LDS;
   },
 
-  clear(): Promise {
-    return RNStorage.clear();
+  async getRawStorage(): Promise {
+    const keys = await RNStorage.getAllKeys();
+    const storage = {};
+    const results = await RNStorage.multiGet(keys);
+    results.map((pair) => {
```

Comment:
can use forEach here instead of map cause you're not capturing the returned array.

map returns an array with one element per element in the original array
forEach return nothing.

so forEach is more memory efficient if you're not returning something.

map composes very nicely with filter.